base/ref.go

```diff
@@ -32,9 +32,9 @@ func SetPublishStatus(r repo.Repo, ref *reporef.DatasetRef, published bool) erro
 	return r.PutRef(*ref)
 }
 
-// ToDatasetRef parses the dataset ref and returns it, allowing datasets with no history only
-// if FSI is enabled.
-func ToDatasetRef(path string, r repo.Repo, allowFSI bool) (*reporef.DatasetRef, error) {
+// ToDatasetRef parses the dataset ref and looks it up in the refstore, allows refs with no history
+// TODO(dustmop): In a future change, remove the third parameter from this function
+func ToDatasetRef(path string, r repo.Repo, _ bool) (*reporef.DatasetRef, error) {
```

Comment:
My only complaint here: we're increasing our reliance on `reporef.DatasetRef`, when we should be transitioning towards `dsref.Ref`. That struct doesn't have a `FSIPath` field, and I'd like to keep it that way. Instead I think we should use this function to process the filepath, and store it in the `Path` field with a prefix that indicates it's an FSI path. Off the top of my head, we could use `file://` for this. 

The long term solution for this whole problem is to use a muxed `qfs.Filesystem` interface that knows how to deal with both local & IPFS paths, further removing the need for `if` statements that redirect to FSI, but that's a _long_ ways off.

This PR is still an invaluable intermediate step that it's not worth delaying on, We should follow up in a future PR that switches the return value.

Reply:
The true fix I think is to switch to dscache, and get rid of base.ToDatasetRef entirely. I have a PR waiting in the wings that starts that change. For now, this PR doesn't increase the usage of base.ToDatasetRef, only remove its third parameter.

Having a muxed qfs.Filesystem would be good, as noted in https://github.com/qri-io/qri/pull/1222.